docs/aws-sts-provisioning.md

```diff
@@ -0,0 +1,74 @@
+# Provisioning AWS STS Clusters
+
+It is possible to use Hive to provision clusters configured to use Amazon's Security Token Service, where cluster components use short lived credentials that are rotated frequently, and the cluster does not have an admin level AWS credential. This feature was added to the in-cluster OpenShift components in 4.7, see documentation [here](https://docs.openshift.com/container-platform/4.7/authentication/managing_cloud_provider_credentials/cco-mode-sts.html).
+
+At present Hive does not automate the STS setup, rather we assume the user configures STS components manually and provides information to Hive. The following instructions refer to the 'ccoctl' tool coming in OpenShift 4.8 with the Cloud Credential Operator. This tool is not yet released but development is underway [here](https://github.com/openshift/cloud-credential-operator) and can be compiled from source, and used to prepare for provisioning 4.7 clusters.
+
+## Extract Credentials Requests from Desired OpenShift Release Image
+
+These vary from release to release, so be sure to use the same release image here as you plan to use for your ClusterDeployment:
+
+```bash
+$ mkdir credrequests/
+$ oc adm release extract quay.io/openshift-release-dev/ocp-release:4.7.1-x86_64 --credentials-requests --cloud=aws > credrequests/credrequests.yaml
+```
+
+
+## Setup STS Infrastructure
+
+```bash
+$ ccoctl create key-pair
+$ ccoctl create identity-provider --name-prefix mystsprefix --public-key-file serviceaccount-signer.public --region us-east-1
+$ ccoctl create iam-roles --credentials-requests-dir /home/dgoodwin/go/src/github.com/openshift/cloud-credential-operator/sts/credrequests/ --identity-provider-arn arn:aws:iam::125931421481:oidc-provider/dgoodsts2-oidc.s3.us-east-1.amazonaws.com --name-prefix dgoodsts2 --region us-east-1
```

Comment:
these commands are not consistent wrt to name prefix and the resources that are being provided.
Can we generalize them?

also `125931421481` sounds like a real AWS account id, let' not use that.